collection_test.go

```diff
@@ -420,6 +421,75 @@ func TestCollection_Count(t *testing.T) {
 	}
 }
 
+func TestCollection_RemoveDocument(t *testing.T) {
+	// Create persistent collection
+	tmpdir, err := os.MkdirTemp(os.TempDir(), "chromem-test-*")
+	if err != nil {
+		t.Fatal("expected no error, got", err)
+	}
+	db, err := NewPersistentDB(tmpdir, false)
+	if err != nil {
+		t.Fatal("expected no error, got", err)
+	}
+	name := "test"
+	metadata := map[string]string{"foo": "bar"}
+	vectors := []float32{-0.40824828, 0.40824828, 0.81649655} // normalized version of `{-0.1, 0.1, 0.2}`
+	embeddingFunc := func(_ context.Context, _ string) ([]float32, error) {
+		return vectors, nil
+	}
+	c, err := db.CreateCollection(name, metadata, embeddingFunc)
+	if err != nil {
+		t.Fatal("expected no error, got", err)
+	}
+	if c == nil {
+		t.Fatal("expected collection, got nil")
+	}
+
+	// Add documents
+	ids := []string{"1", "2"}
+	metadatas := []map[string]string{{"foo": "bar"}, {"a": "b"}}
+	contents := []string{"hello world", "hallo welt"}
+	err = c.Add(context.Background(), ids, nil, metadatas, contents)
+	if err != nil {
+		t.Fatal("expected nil, got", err)
+	}
+
+	// Check count
+	if c.Count() != 2 {
+		t.Fatal("expected 2, got", c.Count())
+	}
+
+	// Check number of files in the persist directory
+	d, err := os.ReadDir(c.persistDirectory)
+	if err != nil {
+		t.Fatal("expected nil, got", err)
+	}
+	if len(d) != 3 { // 2 documents + 1 metadata file
+		t.Fatal("expected 2 files in persist_dir, got", len(d))
+	}
+
+	// Remove document
+	err = c.RemoveDocument(context.Background(), "1")
+	if err != nil {
+		t.Fatal("expected nil, got", err)
+	}
+
+	// Check count
+	if c.Count() != 1 {
+		t.Fatal("expected 1, got", c.Count())
+	}
+
+	// Check number of files in the persist directory
+	d, err = os.ReadDir(c.persistDirectory)
+	if err != nil {
+		t.Fatal("expected nil, got", err)
+	}
+	if len(d) != 2 { // 1 document + 1 metadata file
+		t.Fatal("expected 1 file in persist_dir, got", len(d))
```

Comment:
While we only expect 1 document file, the error could mention both files, as it checks for both.
```suggestion
		t.Fatal("expected 2 file in persist_dir, got", len(d))
```